htdocs/core/lib/files.lib.php

```diff
@@ -2840,8 +2840,14 @@ function dol_compress_dir($inputdir, $outputfile, $mode = "zip", $excludefiles =
  */
 function dol_most_recent_file($dir, $regexfilter = '', $excludefilter = array('(\.meta|_preview.*\.png)$', '^\.'), $nohook = 0, $mode = 0)
 {
+	global $conf;
 	$tmparray = dol_dir_list($dir, 'files', 0, $regexfilter, $excludefilter, 'date', SORT_DESC, $mode, $nohook);
-	return isset($tmparray[0]) ? $tmparray[0] : null;
+	//return isset($tmparray[0])?$tmparray[0]:null;
+	if (!empty($conf->global->MAIN_EMAIL_ATTACH_ALL_FILES)) {
+		return isset($tmparray) ? $tmparray : null;
```

Comment:
dol_most_recent_file may be used on a lot of different use case.
If you change the behaviour according to one option (for email attachment), it will change everywhere including the case where the behaviour should no change.
It seems you try to change the behaviour of a function by setting a global parameter to retrieve it inside the function. Such a need should be addressed by propagating paramater controled by the caller and notby setting a global setup option.